src/saveload/saveload.h

```diff
@@ -873,7 +873,8 @@ static inline bool IsNumericType(VarType conv)
  */
 static inline void *GetVariableAddress(const void *object, const SaveLoad *sld)
 {
-	return const_cast<byte *>((const byte*)(sld->global ? nullptr : object) + (ptrdiff_t)sld->address);
+	if (sld->global) return sld->address;
+	return const_cast<byte *>((const byte *)object + (ptrdiff_t)sld->address);
```

Comment:
```suggestion
	assert(object != nullptr);
	return const_cast<byte *>((const byte *)object + (ptrdiff_t)sld->address);
```
then the other commit is not needed